goblin-breakdown/readme.md

```diff
@@ -0,0 +1,80 @@
+# Goblin Breakdown
+
+Use AI to break down coding problems into smaller pieces.
+
+<!--{{<note type="tip" title="Prompt">}}>-->
+
+If Goblin is offline, use any LLM with the pre-prompt "Break down this task into single responsibility functions".
+
+<!--{{</note>}}-->
+
+## Learning Objectives
+
+```objectives
+- [ ] Use AI to break down a coding problem
+- [ ] Write comments to structure small single responsibility functions
+- [ ] Produce a planned solution to a coding problem
+```
+
+## Requirements
+
+We're practising breaking down our problems again. It must be Saturday!
+
+This time we're going to use AI to help us. [Goblin](https://goblin.tools) is a small, clever integration with ChatGPT that makes magic todo lists. We're going to use it to break down our problem and plan our solution.
+
+### 🧰 1. Setup
+
+You should already be set up but just in case, you need to open this repository in VS Code.
+
+<!--{{<note type="activity" title="Getting Set Up">}}>-->
+
+1.  Set a whole class [timer for 5 minutes](https://www.google.com/search?q=timer+for+5+minutes).
+1.  Fork this repository to your own GitHub account if you have not already.
+1.  Clone the repository to your machine.
+1.  Open in VS Code and go to the `goblin-breakdown` folder.
+
+<!--{{</note>}}-->
+
+## 👣 2. Break down coding problems
+
+Let's try breaking down a coding problem. Let's do a sample problem together. Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Planning a Solution 10m">}}>-->
+
+### 🧱 3. Plan a solution
+
+> Format a UK address and UK phone number from a JSON object and show it as HTML on the page
```

Comment:
I think we should explicitly call out the schema (which can be as simple as saying "There is an example of the JSON returned from the API in `data.json`

I think "formatting an address" when you don't know whether the address comes as one string, or a string per line, or as even more structured fields, is unspecified for this particular workshop in a way which probably hinders more than it helps :)

Reply:
So, interestingly, if you prompt the tool with the json, the steps given are much more vague. 

format the address and phone number and show them as html on a page { "name": "Mahan Esfahani", "address": "789 Pine St, Sometown, Somewhere", "phone": "+44 7700 999123" }

Gives you more like "Format the address and phone number in the desired output format"

If you use the vaguer framing, it often suggests formatting, like +44 to (0)